daemon/api_users.go

```diff
@@ -208,14 +210,54 @@ func logoutUser(c *Command, r *http.Request, user *auth.UserState) Response {
 	return SyncResponse(nil, nil)
 }
 
+// this might need to become a function, if having user admin becomes a config option
+var hasUserAdmin = !release.OnClassic
+
+const noUserAdmin = "system user administration via snapd is not allowed on this system"
+
+func postUsers(c *Command, r *http.Request, user *auth.UserState) Response {
+	if !hasUserAdmin {
+		return MethodNotAllowed(noUserAdmin, r.Method)
+	}
+
+	var postData postUserData
+
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&postData); err != nil {
+		return BadRequest("cannot decode user action data from request body: %v", err)
+	}
+	if decoder.More() {
+		return BadRequest("spurious content after user action")
+	}
+	switch postData.Action {
+	case "create":
+		return createUser(c, postData.postUserCreateData)
+	case "remove":
+		return removeUser(c, postData.Username)
```

Comment:
Just a comment as you said more is coming in the follow up. I would go for an option structure for user removal as there may be things that we'd like to add later (e.g. backup-home: true, or something like that) and now it's easy and cheap to include this.